app/services/time_discrepancy_loader.rb

```diff
@@ -5,28 +5,25 @@ class TimeDiscrepancyLoader
 
   # `scope` is an AR collection proxy for TimerSession
   def self.uneven_timer_sessions(scope)
-    scope.joins(:time_entries)
-         .group(:id, :timer_start, :timer_end)
-         .having(
-           <<~SQL
-             CAST(SUM(time_entries.hours)
-             AS DECIMAL(10, #{DECIMALS_TO_ROUND_TO}))
-             !=
-             CAST((#{time_difference_expr}) / 3600.0
-             AS DECIMAL(10, #{DECIMALS_TO_ROUND_TO}))
-           SQL
-         )
-  end
-
-  def self.time_difference_expr
-    if using_postgresql?
-      'EXTRACT(EPOCH FROM timer_sessions.timer_end) - EXTRACT(EPOCH FROM timer_sessions.timer_start)'
-    else
-      'TIMESTAMPDIFF(SECOND, timer_sessions.timer_start, timer_sessions.timer_end)'
-    end
+    scope.joins(timer_session_time_entries: :time_entry)
+         .group('timer_sessions.id')
+         .having("#{round('timer_sessions.hours')} != #{round('SUM(time_entries.hours)')}")
   end
 
   def self.using_postgresql?
     ActiveRecord::Base.connection.adapter_name == 'PostgreSQL'
+  def self.round(attribute)
+    "CAST(#{attribute} AS DECIMAL(10, #{DECIMALS_TO_ROUND_TO}))"
   end
```

Comment:
This method is dangerous because it could be called with non-numeric fields containing SQL injectable code. Please make it a private method at least.